src/codegen/sys/cargo_toml.rs

```diff
@@ -46,15 +46,31 @@ fn fill_empty(root: &mut Table, env: &Env, crate_name: &str) {
 
     let deps = upsert_table(root, "dependencies");
     for ext_lib in &env.config.external_libraries {
+        let dep = upsert_table(deps, &ext_lib.crate_name);
         let ext_package = if ext_lib.crate_name == "cairo" {
             format!("{}-sys-rs", ext_lib.crate_name)
         } else if ext_lib.crate_name == "gdk_pixbuf" {
             "gdk-pixbuf-sys".into()
         } else {
             format!("{}-sys", ext_lib.crate_name)
         };
-        let dep = upsert_table(deps, &*ext_package);
-        set_string(dep, "git", "https://github.com/gtk-rs/gtk-rs-core");
+        let repo_url = match ext_package.as_str() {
+            // Not sure if glib-macros and gobject-sys are needed here
+            "cairo-sys-rs" | "gdk-pixbuf-sys" | "gio-sys" | "gobject-sys" | "glib-macros"
```

Comment:
That's pretty bad because it means gir can only generate crates it knows about. Which is not what gir is aiming for...

Reply:
Maybe I am mistaken, but I think the current code always adds `git = "https://github.com/gtk-rs/gtk-rs-core"` to all dependencies in the Cargo.toml. This is only correct for the crates included in that repo. For all other dependencies (including e.g. gdk, gtk3) the build will fail because the url is wrong and the user needs to look up the url under which the crate can be found and correct it.
With this code change, at least for the current dependencies from the gtk-rs-core, gtk3-rs and gtk4-rs repos, it uses the correct links. I guess the `&_ => "Unknown dependency, please create an issue",` branch could be changed to return the gtk-rs-core url. Since I believe I added all dependencies included in the repos, I don't see how that would be better though since the dependency would not be found under that url.
I don't know how gir would be able to determine the correct url for arbitrary dependencies. If it is a dependency that is not part of gtk-rs repos, the user can correct the url. If it is a new crate that is part of any of the gtk-rs repos, the first user depending on it would get a build error. They would see the compiler complaining about the dependency not being found and read the part asking them to open an issue. Ideally they would quickly create an issue, the dependency is added to the match statement and it's fixed. If they don't open an issue, they could easily fix the broken url in the Cargo.toml file themselves and somebody else would raise the issue.

Reply:
I guess we should write "ADD GIT REPOSITORY URL HERE".

Reply:
Here is an example:
Before my proposed code change this would be added to the Cargo.toml of the sys crate:

[dependencies.gtk-sys]
git = "https://github.com/gtk-rs/gtk-rs-core"

With the proposed code change this is added to the Cargo.toml of the sys crate:

[dependencies.gtk]
package = "gtk-sys"
git = "https://github.com/gtk-rs/gtk3-rs"

Reply:
And with my suggestion it would give:

```
[dependencies.gtk]
package = "gtk-sys"
git = "ADD GIT REPOSITORY URL HERE"
```

Reply:
> That's pretty bad because it means gir can only generate crates it knows about. Which is not what gir is aiming for...

Is it ? In general most of the libraries are based on glib/gobject and then either gstreamer/gtk3/gtk4. but for other cases it is better to write some fallback like
```
[dependencies.gtk]
package = "gtk-sys"
git = "ADD GIT REPOSITORY URL HERE"
```

Reply:
Then maybe we should only list `gtk-rs-core` crates?

Reply:
I added the gstreamer crates and its url and changed the message for unknown dependencies. This minimizes the manual changes users need to do.